tools/RAiDER/models/gmao.py

```diff
@@ -0,0 +1,171 @@
+import datetime
+import numpy as np
+import os
+from pyproj import CRS
+import re
+
+from RAiDER.models.weatherModel import WeatherModel
+
+
+def Model():
+    return GMAO()
+
+class GMAO(WeatherModel):
+    # I took this from GMAO model level weblink
+    # https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv
+    def __init__(self):
+        # initialize a weather model
+        WeatherModel.__init__(self)
+
+        self._humidityType = 'q'
+        self._model_level_type = 'ml' # Default, pressure levels are 'pl'
+        
+        self._classname = 'gmao'
+        self._dataset = 'gmao'
+
+        # Tuple of min/max years where data is available. 
+        self._valid_range = (datetime.datetime(2017,12,1),"Present")
+        self._lag_time = datetime.timedelta(hours=0.3125) # Availability lag time in days
+
+        # model constants
+        self._k1 = 0.776  # [K/Pa]
+        self._k2 = 0.233 # [K/Pa]
+        self._k3 = 3.75e3 # [K^2/Pa]
+
+        # horizontal grid spacing
+        self._lat_res = 0.25
+        self._lon_res = 0.3125
+        self._x_res = 0.3125
+        self._y_res = 0.25
+
+        self._Name = 'GMAO'
+        self._files = None
+        self._bounds = None
+
+        # Projection
+        self._proj = CRS.from_epsg(4326)
+
+    
+    
+    
+    def _fetch(self, lats, lons, time, out, Nextra = 2):
+        '''
+        Fetch weather model data from GMAO: note we only extract the lat/lon bounds for this weather model; fetching data is not needed here as we don't actually download any data using OpenDAP
+        '''
+        # bounding box plus a buffer
+        lat_min, lat_max, lon_min, lon_max = self._get_ll_bounds(lats, lons, Nextra)
+        self._bounds = (lat_min, lat_max, lon_min, lon_max)
+
+
+    
+    
+
+    def load_weather(self,f):
+        '''
+        Consistent class method to be implemented across all weather model types.
+        As a result of calling this method, all of the variables (x, y, z, p, q,
+        t, wet_refractivity, hydrostatic refractivity, e) should be fully
+        populated.
+        '''
+        self._load_model_level(f)
+
+
+    
+    def _load_model_level(self,f):
+        '''
+        Get the variables from the GMAO link using OpenDAP
+        '''
+        # calculate the array indices for slicing the GMAO variable arrays
+        lat_min_ind = int((self._bounds[0] - (-90.0)) / self._lat_res)
+        lat_max_ind = int((self._bounds[1] - (-90.0)) / self._lat_res)
+        lon_min_ind = int((self._bounds[2] - (-180.0)) / self._lon_res)
+        lon_max_ind = int((self._bounds[3] - (-180.0)) / self._lon_res)
+        
+        import datetime as dt
```

Comment:
It's already imported at the top of the file. IMO you should just do:
```python3
from datetime import datetime
```
once at the top of the file.

Reply:
You also need to update the places where you're using it.